JustSaying/Messaging/MessageHandling/IMessageContextReader.cs

```diff
@@ -0,0 +1,7 @@
+namespace JustSaying.Messaging.MessageHandling
+{
+    public interface IMessageContextReader
+    {
+        MessageContext Read();
```

Comment:
would it be `ReadMessageContext` or `ReadFromAsyncLocal` or ?

Reply:
If we're making this model `IHttpContextAccessor`, why isn't it just a simple property with a get and a set? As it stands we've got an interface that's a front to a static object (which doesn't fix the testability issues because it's still a shared static) which has an internal set method.

Reply:
A mockable "Get" allows you to supply fake data to a handler (test not yet written)
Point taken that also adding a mockable "set" also allows you to inspect what's happening in integration / localstack scenarios.

Reply:
Also, they're doing some fancy object indirection trick there. Should we be doing the same?

Reply:
I think @shaynevanasperen means `private class HttpContextHolder` [here](https://github.com/aspnet/HttpAbstractions/blob/master/src/Microsoft.AspNetCore.Http/HttpContextAccessor.cs). 
I don't know of any benefit of adding that yet?